transcendence-app/src/user/user.service.ts

```diff
@@ -46,12 +47,22 @@ export class UserService {
     return users ? users.map((user) => new User(user)) : null;
   }
 
-  async retrieveUserWithUserName(username: string): Promise<User | null> {
+  async retrieveUserWithUserName(
+    username: string,
+    userMe?: User,
+  ): Promise<UserDto | null> {
     const user = await this.userRepository.getByUsername(username);
-    return user ? new User(user) : null;
+
+    if (user && userMe) {
+      return {
+        ...user,
+        isBlocked: await this.isUserBlockedByMe(userMe, user.id),
+      };
+    }
```

Comment:
Fixed to return an instance! that being said:
- As explained in above comment I don't think is a good idea to add `isBlocked` to `User` domain
- I didn't add any serialization annotation because the data could come from a `User` or a `UserEntity` which they are already serialized. Not 100% sure about this btw, let me now and we can fix it in a future task

Thanks as always for your comments! :smiley:

Reply:
Forgot to push the commit with the change :sweat_smile: . Check -> https://github.com/42AGV/ft_transcendence/pull/288/files